pydm/data_plugins/plugin.py

```diff
@@ -161,26 +161,32 @@ def __init__(self):
     def get_address(channel):
         return protocol_and_address(channel.address)[1]
 
+    @staticmethod
+    def get_connection_id(channel):
+        return PyDMPlugin.get_address(channel)
+
     def add_connection(self, channel):
         with self.lock:
-            address = self.get_address(channel)
             # If this channel is already connected to this plugin lets ignore
             if channel in self.channels:
                 return
+            id = self.get_connection_id(channel)
```

Comment:
Careful with `id` as a variable - it's a builtin function.

Reply:
Addressed via https://github.com/slaclab/pydm/pull/637/commits/54f602d9fd2b126a2dd5769222bac126812f3684